src/contexts/layout-context.tsx

```diff
@@ -0,0 +1,31 @@
+'use client'
+import * as React from 'react'
+
+interface LayoutContext {
+  isSidebarOpen: boolean
+  sidebarOpenToggle: () => void
+}
+
+const LayoutContext = React.createContext<LayoutContext | undefined>(undefined)
+
+const LayoutContextProvider = ({ children }: { children: React.ReactNode }) => {
+  const [isSidebarOpen, setIsSidebarOpen] = React.useState(false)
```

Comment:
I think it will be more efficient to direct import `useState` function instead import all of React 
```suggestion
  const [isSidebarOpen, setIsSidebarOpen] = useState(false)
```